src/devices/Arduino/samples/Arduino.sample.cs

```diff
@@ -0,0 +1,506 @@
+            if (args.Length > 1)
+            {
+                portName = args[1];
+            }
+
+            TextWriter logFile = new StringWriter();
+
+            using (var port = new SerialPort(portName, 115200))
+            {
+                Console.WriteLine($"Connecting to Arduino on {portName}");
+                try
+                {
+                    port.Open();
+                }
+                catch (UnauthorizedAccessException x)
+                {
+                    Console.WriteLine($"Could not open COM port: {x.Message} Possible reason: Arduino IDE connected or serial console open");
+                    return;
+                }
+
+                DebugLogStream dls = new DebugLogStream(port.BaseStream, logFile);
+
+                ArduinoBoard board = new ArduinoBoard(dls);
+                try
+                {
+                    board.LogMessages += BoardOnLogMessages;
+                    board.Initialize();
+                    Console.WriteLine($"Connection successful. Firmware version: {board.FirmwareVersion}, Builder: {board.FirmwareName}");
+                    while (Menu(board))
+                    {
+                    }
+                }
+                catch (TimeoutException x)
+                {
+                    Console.WriteLine($"No answer from board: {x.Message} ");
+                }
+                finally
+                {
+                    port.Close();
+                    board?.Dispose();
+                }
+            }
+
+            Debug.Write(logFile.ToString());
+        }
+
+        private static void BoardOnLogMessages(string message, Exception? exception)
+        {
+            Console.WriteLine("Log message: " + message);
+            if (exception != null)
+            {
+                Console.WriteLine(exception);
+            }
+        }
+
+        private static bool Menu(ArduinoBoard board)
+        {
+            Console.WriteLine("Hello I2C and GPIO on Arduino!");
+            Console.WriteLine("Select the test you want to run:");
+            Console.WriteLine(" 1 Run I2C tests with a BMP280");
+            Console.WriteLine(" 2 Run GPIO tests with a simple led blinking on GPIO6 port");
+            Console.WriteLine(" 3 Run polling button test on GPIO2");
+            Console.WriteLine(" 4 Run event wait test event on GPIO2 on Falling and Rising");
+            Console.WriteLine(" 5 Run callback event test on GPIO2");
+            Console.WriteLine(" 6 Run PWM test with a LED dimming on GPIO6 port");
+            Console.WriteLine(" 7 Blink the LED according to the input on A1");
+            Console.WriteLine(" 8 Read analog channel as fast as possible");
+            Console.WriteLine(" 9 Run SPI tests with an MCP3008 (experimental)");
+            Console.WriteLine(" 0 Detect all devices on the I2C bus");
+            Console.WriteLine(" H Read DHT11 Humidity sensor on GPIO 3 (experimental)");
+            Console.WriteLine(" X Exit");
```

Comment:
I'd suggest simplifying this and creating separate projects instead (at least for sensors, GPIO is ok to be joint into one)

Reply:
I think that won't make things simpler, especially also for maintaining them.